### PR TITLE
Addresses Issue 240: allowing choice of normalization type

### PR DESCRIPTION
**Pull request by Zach-ER**
## Status
READY

## Description
Changes `with_bn` to a `bn_type`: allows different normalization techniques, set with strings. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Todos
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by tvercaut** [comment]
I find the variable name `bn_type` confusing if you end up not using BN... Maybe replace it with `featnorm_type` or something along these lines?

**Comment by Zach-ER** [comment]
Yeah, I agree that `featnorm_type` is a better descriptor.

**Comment by Zach-ER** [comment]
@tvercaut changed as per suggestion, @wyli now ready to merge i think

**Comment by fepegar** [comment]
For the sake of clarity, why not calling the variable `feature_normalization_type`?

**Comment by tvercaut** [comment]
`feature_normalization_type` might be somehow overly verbose given that `featnorm` is unambiguous, but I guess it would be good to start having naming conventions to refer to as part of NiftyNet.

As TF uses the wording `batch_normalization` (https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization), we could also use  `feature_normalization` for the flag name. Not sure if the word `type` adds anything in the flag name.
